File: chaerim/Programmers/Level_2/Lv2_영어_끝말잇기.py
def solution(n, words):
    answer = []

    word_list = []
    word_list.append(words[0])
    for i in range(1, len(words)):
        if words[i] in word_list or words[i][0] != words[i - 1][-1]:
            answer = [(i%n) + 1, (i//n) + 1]
            break
        word_list.append(words[i])

    if not answer:
        answer = [0, 0]

    return answer

File: seongwon/Programmers/Lv2_영어_끝말잇기.java
class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];
        int len = words.length; 
        boolean[] visited = new boolean[len];
        char lastChar = words[0].charAt(words[0].length() - 1); 
        visited[0] = true;

        for (int i = 1; i < len; i++) {
            if (visited[i]) {
                answer[0] = (i % n) + 1;
                answer[1] = (i / n) + 1;
                return answer;
            }

            if (lastChar != words[i].charAt(0)) {
                answer[0] = (i % n) + 1;
                answer[1] = (i / n) + 1;
                return answer;
            }

            if (isDuplicate(words, i)) {
                answer[0] = (i % n) + 1;
                answer[1] = (i / n) + 1;
                return answer;
            }
            
            visited[i] = true;
            lastChar = words[i].charAt(words[i].length() - 1);
        }

        answer[0] = 0;
        answer[1] = 0;
        return answer;
    }

    private boolean isDuplicate(String[] words, int index) {
        for (int i = 0; i < index; i++) {
            if (words[i].equals(words[index])) {
                return true;
            }
        }
        return false;
    }
}
